.github/workflows/ci.yaml

```diff
@@ -22,3 +22,43 @@ jobs:
     steps:
     - uses: actions/checkout@v1
     - run: "cd tests && make check-component-features"
+
+# Kubernetes
+  build-vector-kubernetes:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build
+      run: cargo build --release --no-default-features  --features "sources-kubernetes transforms-kubernetes sinks-console"
```

Comment:
It would be nice if we could build Vector once for all of our tests.

Reply:
I think in general we should have a check job that just runs check which should be fast and good for quick feedback.

Then one test job that tests everything, hopefully we won't run into noise issues with running kube and all our docker containers together. This then would allow us to build vector once in test mode and cache those build artifacts.

Reply:
> It would be nice if we could build Vector once for all of our tests, if possible.

Yep, sharing the binaries could really speed up the test harness execution. We probably should address this as a part of our Github Actions switch. We'll have an opportunity to apply some grand design to our CI flows.

> I think in general we should have a check job that just runs check which should be fast and good for quick feedback.

I have some ideas on how to organize the jobs/steps/actions/workflows to make them much more compact and composable that what we have at Circle CI. And more maintainable, and efficient too.

Reply:
> Yep, sharing the binaries could really speed up the test harness execution. We probably should address this as a part of our Github Actions switch.

I'll see if it can be achieved here, because it would half the execution time from 40min to 20min, and half the total work.